### PR TITLE
Task 5 (Integration with S3)

### PR DESCRIPTION
**Pull request by shall-it**
Task 5.1
New service import-service was created; S3 bucket with folder 'uploaded' was created manually into AWS console

Task 5.2
API Gateway endpoint: https://sw4mnq00z4.execute-api.us-east-1.amazonaws.com/prod
importProductsFile: https://sw4mnq00z4.execute-api.us-east-1.amazonaws.com/prod/import
Please use the URL https://dyfk99rjrorkr.cloudfront.net/admin/products to upload the CSV file from the frontend
CSV file will be uploaded into 'uploaded' folder of S3 bucket via presigned URL

Step 1

Step 2

Step 3

Task 5.3
importFileParser is triggered by s3:ObjectCreated:* event only from 'uploaded' folder of S3 bucket; lambda uses readable stream

Step 4

Additional tasks are also completed (unit tests for both lambdas, moving the file from the uploaded folder into the parsed folder)

Step5

Frontend application is integrated with Import Service (/import API) and represented as ability to upload CSV file into S3 bucket: https://dyfk99rjrorkr.cloudfront.net/admin/products (PR https://github.com/shall-it/nodejs-aws-shop-react/pull/4)
Common Frontend URL: https://dyfk99rjrorkr.cloudfront.net

### CODE REVIEW

File: import-service/cdk.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  runtime: lambda.Runtime.NODEJS_20_X,
+  environment: {
+    IMPORT_AWS_REGION: importAWSRegion,
+    BUCKET_NAME: bucketName
+  }
+};
+
+const bucket = s3.Bucket.fromBucketName(stack, 'importBucket', bucketName);
+
+const parsedDirectory = 'parsed/';
```
Comment (SAK74) [inline import-service/cdk.ts]: Doubled var (line 30 *prefix_parsed*) :)

File: import-service/src/handlers/importProductsFile.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { buildResponse } from '../utils';
+import { S3 } from 'aws-sdk';
+
+export const handler = async (event: any) => {
+    const s3 = new S3();
+    const bucketName = process.env.BUCKET_NAME!;
+    const filePath = event.queryStringParameters.name;
+
+    if (!bucketName) {
+        return buildResponse(500,
```
Comment (SAK74) [inline import-service/src/handlers/importProductsFile.ts:10]: This shouldn't be in response rather, whereas fileName validation   would be appreciated

File: import-service/src/handlers/importFileParser.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import { buildResponse } from '../utils';
+import { S3 } from 'aws-sdk';
+import csv from 'csv-parser';
+import { Readable } from 'stream';
+
+const s3 = new S3();
+const bucketName = process.env.BUCKET_NAME!;
+
+export const handler = async (event: any) => {
```
Comment (SAK74) [inline import-service/src/handlers/importFileParser.ts]: **any** is very important bad habit to my mind!

File: import-service/src/handlers/importFileParser.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { buildResponse } from '../utils';
+import { S3 } from 'aws-sdk';
+import csv from 'csv-parser';
+import { Readable } from 'stream';
+
+const s3 = new S3();
+const bucketName = process.env.BUCKET_NAME!;
```
Comment (SAK74) [inline import-service/src/handlers/importFileParser.ts]: *bucketName* you can receive from event...

File: import-service/src/handlers/importProductsFile.ts (hunk 2 above)
Comment (shall-it) [reply import-service/src/handlers/importProductsFile.ts:10]: Serhiy, I've left comment under Cross-Check: Submit review

**Pull request merged**